### PR TITLE
extraction-3-create-1st-transform-add-missing-jsx-attr-and-initial-codemod-setup - CTR-20

### PR DESCRIPTION
**Pull request by kiprasmel**
how to review: https://github.com/pipedrive/CodeshiftCommunity/pull/9 (commit-by-commit is still best)

### CODE REVIEW

File: community/@pipedrive__convention-ui-react/src/5.0.0/config.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const cuiLibraryName = "@pipedrive/convention-ui-react" as const; //  TODO EXTRACT. atm only used in specific configs so that's fine.
```
Comment (martikaljuve) [inline community/@pipedrive__convention-ui-react/src/5.0.0/config.ts:1]: remove comment.

File: packages/reusable-transforms/src/add-missing-jsx-attribute/add-missing-jsx-attribute.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// import { toComment, tick, parens } from "nice-comment";
+
+// import { insertCommentBefore } from "@codeshift/utils";
+
+// import { inlineCommentPrefix } from "../config";
+
+import { getParsedImports } from "../utils/getParsedImports";
+import { Transformer } from "../utils/Transformer";
+
+export interface ConfigToAddMissingJsxProp {
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/add-missing-jsx-attribute/add-missing-jsx-attribute.ts:11]: Add example as comment

File: packages/reusable-transforms/src/utils/getParsedImports.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    from: string;
+
+    /** name used in end user's code */
+    isImportedAs: string;
+} & (
+    | {
+          isDefaultImport: true;
+
+          /** original name (we don't know) */
+          wasExportedAs?: null; // TODO `never`
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/utils/getParsedImports.ts:28]: apply TODO?

File: packages/reusable-transforms/src/utils/getParsedImports.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+          isDefaultImport: false;
+
+          /** original name */
+          wasExportedAs: string;
+      }
+);
+
+// const unsupportedNonLiteralRequireMsg = "un-supported non-literal require: " as const;
+
+export const getParsedImports = <T = any>(
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/utils/getParsedImports.ts:40]: Maybe could add an example link to AST Explorer in a JSDoc?

File: packages/reusable-transforms/src/utils/getParsedImports.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        }
+
+        const from: string = node.source.value;
+
+        if (!node.specifiers || !node.specifiers.length) {
+            return;
+        }
+
+        node.specifiers.forEach((specifier): void => {
+            if (j.ImportDefaultSpecifier.check(specifier)) {
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/utils/getParsedImports.ts:66]: Would be useful to have examples as comments for the different branches here.

File: packages/reusable-transforms/src/utils/getParsedImports.ts [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                }
+
+                ret.push({
+                    from,
+                    isDefaultImport: false,
+                    wasExportedAs: specifier.imported.name,
+                    isImportedAs: specifier.local.name,
+                });
+            } else {
+                /**
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/utils/getParsedImports.ts:89]: `import *`?

File: packages/reusable-transforms/src/utils/Transformer.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+             */
+            (configs as typeof transformer extends Transformer<infer C> ? C[] : never).map(
+				config =>
+                transformer(
+                    // api.j, //
+                    // source,
+                    // file,
+                    // api,
+                    // options,
+					...transformerParamsExceptConfig,
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/utils/Transformer.ts:55]: formatting?

**Review by martikaljuve — CHANGES_REQUESTED** [review]
.

File: packages/reusable-transforms/src/add-missing-jsx-attribute/add-missing-jsx-attribute.ts (hunk 2 above)
Comment (kiprasmel) [reply packages/reusable-transforms/src/add-missing-jsx-attribute/add-missing-jsx-attribute.ts:52]: added.

File: packages/reusable-transforms/src/utils/getParsedImports.ts (hunk 4 above)
Comment (kiprasmel) [reply packages/reusable-transforms/src/utils/getParsedImports.ts:53]: added.

File: packages/reusable-transforms/src/utils/Transformer.ts (hunk 7 above)
Comment (kiprasmel) [reply packages/reusable-transforms/src/utils/Transformer.ts]: prettier's fault since we didn't set up eslint. won't bother tbh.

**Review by martikaljuve — APPROVED** [review]

**Pull request merged**